operator/config/localdb.go

```diff
@@ -0,0 +1,423 @@
+func (l *LocalDB) apply(configName string) error {
+
+	clusterName := l.configMap.GetObjectMeta().GetLabels()[config.LABEL_PG_CLUSTER]
+	namespace := l.configMap.GetObjectMeta().GetNamespace()
+
+	log.Debugf("Cluster Config: applying local config %s in cluster %s "+
+		"(namespace %s)", configName, clusterName, namespace)
+
+	localConfig, err := l.getLocalConfig(configName)
+	if err != nil {
+		return err
+	}
+
+	// selector in the format "pg-cluster=<cluster-name>,deployment-name=<server-name>"
+	selector := fmt.Sprintf("%s=%s,%s=%s", config.LABEL_PG_CLUSTER, clusterName,
+		config.LABEL_DEPLOYMENT_NAME, strings.TrimSuffix(configName, pghLocalConfigSuffix))
+	dbPodList, err := l.kubeclientset.CoreV1().Pods(namespace).List(metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return err
+	}
+	dbPod := &dbPodList.Items[0]
+
+	// add the config name and patroni port as params for the call to the apply & reload script
+	applyCommand := append(applyAndReloadConfigCMD, localConfig, config.DEFAULT_PATRONI_PORT)
+
+	stdout, stderr, err := kubeapi.ExecToPodThroughAPI(l.restConfig, l.kubeclientset, applyCommand,
+		dbPod.Spec.Containers[0].Name, dbPod.GetName(), namespace, nil)
+
+	if err != nil {
+		log.Error(stderr, stdout)
+		return err
+	}
+
+	log.Debugf("Cluster Config: successfully applied local config %s in cluster %s "+
+		"(namespace %s)", configName, clusterName, namespace)
+
+	return nil
+}
+
+// clean removes any local database server configurations from the configMap included in the
+// LocalDB if the database server they are associated with no longer exists
+func (l *LocalDB) clean() error {
+
+	var jsonPatch = []util.JSONPatchOperation{}
+	var cmlocalConfigs []string
+
+	// first grab all current local configs from the configMap
+	for configName := range l.configMap.Data {
+		if strings.HasSuffix(configName, pghaLocalConfigSuffix) {
+			cmlocalConfigs = append(cmlocalConfigs, configName)
+		}
+	}
+
+	// now see if any need to be deleted
+	for _, cmLocalConfig := range cmlocalConfigs {
+		deleteConfig := true
+		for _, managedConfigName := range l.configNames {
+			if cmLocalConfig == managedConfigName {
+				deleteConfig = false
+				break
+			}
+		}
+		if deleteConfig {
+			jsonPatch = append(jsonPatch, util.JSONPatchOperation{
+				Op:   "remove",
+				Path: fmt.Sprintf("/data/%s", cmLocalConfig),
+			})
+
+		}
+	}
+
+	jsonOpBytes, err := json.Marshal(jsonPatch)
+	if err != nil {
+		return err
+	}
+
+	if _, err := l.kubeclientset.CoreV1().ConfigMaps(l.configMap.GetNamespace()).Patch(
+		l.configMap.GetName(), types.JSONPatchType, jsonOpBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getLocalConfigFromCluster obtains the local configuration for a specific database server in the
+// cluster.  It also returns the Pod that is currently running that specific server.
+func (l *LocalDB) getLocalConfigFromCluster(configName string) (*LocalDBConfig,
+	*corev1.Pod, error) {
```

Comment:
Do we need to return the Pod? We don't seem to be using it in the other calls.

Reply:
Thanks, that was leftover from an earlier iteration of this thing.

The code has been updated to no longer return the Pod.